src/application.c

```diff
@@ -256,6 +257,21 @@ void copy_clicked_handler(GtkWidget *widget, struct swappy_state *state) {
   clipboard_copy_drawing_area_to_selection(state);
 }
 
+void delete_tmpfile(struct swappy_state *state) {
+  if (g_unlink(state->file_str) != 0)
```

Comment:
This works when using `-f -` command but will also delete a non temporary file when using:

```
./build/swappy -f docs/images/screenshot-1.0.0.png
```

I'm working on a better fix. Thanks for bringing this up @apiraino